session.rst

```diff
@@ -1,15 +1,276 @@
+subsystem which is designed to provide session management that you can use to
+store information about the user between requests through a clear
+object-oriented interface using a variety of session storage drivers.
+
+Symfony sessions are designed to replace the usage of the ``$_SESSION`` super
+global and native PHP functions related to manipulating the session like
+``session_start()``, ``session_regenerate_id()``, ``session_id()``,
+``session_name()``, and ``session_destroy()``.
+
+.. note::
+
+    Sessions are only started if you read or write from it.
+
+Installation
+------------
+
+You need to install the HttpFoundation component to handle sessions:
+
+.. code-block:: terminal
+
+    $ composer require symfony/http-foundation
+
+.. _session-intro:
+
+Basic Usage
+-----------
+
+The session is available through the ``Request`` object and the ``RequestStack``
+service. Symfony injects the ``request_stack`` service in services and controllers
+if you type-hint an argument with :class:`Symfony\\Component\\HttpFoundation\\RequestStack`::
+
+.. configuration-block::
+
+    .. code-block:: php-symfony
+
+        use Symfony\Component\HttpFoundation\RequestStack;
+
+        class SomeService
+        {
+            private $requestStack;
+
+            public function __construct(RequestStack $requestStack)
+            {
+                $this->requestStack = $requestStack;
+
+                // Accessing the session in the constructor is *NOT* recommended, since
+                // it might not be accessible yet or lead to unwanted side-effects
+                // $this->session = $requestStack->getSession();
+            }
+
+            public function someMethod()
+            {
+                $session = $this->requestStack->getSession();
+
+                // ...
+            }
+        }
+
+    .. code-block:: php-standalone
+
+        use Symfony\Component\HttpFoundation\Session\Attribute\AttributeBag;
+        use Symfony\Component\HttpFoundation\Session\Session;
+        use Symfony\Component\HttpFoundation\Session\Storage\NativeSessionStorage;
+
+        $session = new Session();
+        $session->start();
+
+        // Pass a storage and a bag implementation explicitely
+        $session = new Session(new NativeSessionStorage(), new AttributeBag());
+
+From a Symfony controller, type-hint an argument with
+:class:`Symfony\\Component\\HttpFoundation\\RequestStack` or
+:class:`Symfony\\Component\\HttpFoundation\\Request`::
+
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\RequestStack;
+    use Symfony\Component\HttpFoundation\Response;
+
+    public function index(RequestStack $requestStack): Response
+    {
+        $session = $requestStack->getSession();
+
+        // ...
+    }
+
+    public function index(Request $request): Response
+    {
+        $session = $request->getSession();
+
+        // ...
+    }
+
+Attributes
+----------
```

Comment:
Only a minor nitpick, but it might mislead people jumping to this section, thinking this is related to PHP 8 Attributes
```suggestion
Session Attributes
------------------
```

Reply:
Right, fixed